asserts/asserts.go

```diff
@@ -103,7 +103,10 @@ var maxSupportedFormat = map[string]int{}
 
 func init() {
 	// register maxSupportedFormats while breaking initialisation loop
-	maxSupportedFormat[SnapDeclarationType.Name] = 1
+
+	// 1: plugs and slots
+	// 2: support for $SLOT()/$PLUG()/$MISSING
```

Comment:
we don't at the moment have a way to express the constraints that an attribute is omitted, $MISSING constraints the attribute not to be set